File: frontend/RecipeApplication/src/app/Models/Message/message.ts
import { User } from "../User/user";

export class Message {
  id: number;
  sender: User;
  receiver: User;
  content: string;
  timestamp: Date | string;
  wasSeen: boolean;

  constructor(sender: User, receiver: User, content: string, timestamp: Date | string) {
    this.id = -1; 
    this.sender = sender;
    this.receiver = receiver;
    this.content = content;
    this.timestamp = timestamp;
    this.wasSeen = false;
  }
}

File: backend/src/main/java/com/example/recipeapp/Model/Message.java
package com.example.recipeapp.Model;

import com.example.recipeapp.Model.User;
import com.sun.istack.NotNull;
import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "messages")
@Data
public class Message {
    @Id
    @SequenceGenerator(
            name = "messages_sequence",
            sequenceName = "messages_sequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "messages_sequence"
    )
    private long id;

    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    private User sender;

    @ManyToOne
    @JoinColumn(name = "receiver_id", nullable = false)
    private User receiver;

    @NotNull
    @Column(name = "content", length = 16384)
    private String content;

    @Temporal(TemporalType.TIMESTAMP)
    private Date timestamp;

    @Column(name = "was_seen")
    private boolean wasSeen;
}